packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.base.tsx

```diff
@@ -42,13 +47,15 @@ export class CheckboxBase extends React.Component<ICheckboxProps, ICheckboxState
 
     if (process.env.NODE_ENV !== 'production') {
       warnMutuallyExclusive('Checkbox', props, {
-        checked: 'defaultChecked'
+        checked: 'defaultChecked',
+        indeterminate: 'defaultIndeterminate'
       });
     }
 
     this._id = this.props.id || getId('checkbox-');
     this.state = {
-      isChecked: !!(props.checked !== undefined ? props.checked : props.defaultChecked)
+      isChecked: !!(props.checked !== undefined ? props.checked : props.defaultChecked),
+      isIndeterminate: !!(props.defaultIndeterminate ? props.defaultIndeterminate : props.indeterminate)
```

Comment:
These checks should be similar to the ones for checked: `!!(props.indeterminate !== undefined ? props.indeterminate : props.defaultIndeterminate)`. The controlled prop is supposed to take precedence over the uncontrolled one.